### PR TITLE
Fix report flags not reporting on subsequent runs when cache is used

### LINKED ISSUE #5049: Fix report flags not reporting on subsequent runs when cache is used
> Clearly describe the bug

When using the `--rd` switch, if there is a needless disable, the second or any subsequent run doesn't report it.

> Which rule, if any, is the bug related to?

None

> What stylelint configuration is needed to reproduce the bug?

Probably doesn't matter:

```json
{
  "extends": [
    "stylelint-config-twbs-bootstrap/scss"
  ],
  "rules": {
    "function-disallowed-list": [
      "calc",
      "lighten",
      "darken"
    ],
    "property-disallowed-list": [
      "border-radius",
      "border-top-left-radius",
      "border-top-right-radius",
      "border-bottom-right-radius",
      "border-bottom-left-radius",
      "transition"
    ],
    "scss/dollar-variable-default": [
      true,
      {
        "ignore": "local"
      }
    ],
    "scss/selector-no-union-class-name": true
  }
}
```

> Which version of stylelint are you using?

13.7.2 and 13.8.0

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

CLI: `stylelint "**/*.{css,scss}" --cache --cache-location .cache/.stylelintcache --rd`

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

Probably not

> What did you expect to happen?

Any needless disables should be reported on subsequent runs.

> What actually happened (e.g. what warnings or errors did you get)?

No needless disables were reported. **If I delete the cache or remove the `--cache` CLI flag, then the needless disables are reported**.

@jeddy3 I verified this and indeed happens with 13.7.0, 13.7.2 and 13.8.0 that I could quickly test.

### PR DESCRIPTION
**Pull request by ybiquitous**
The bug's reason is that the `stylelintError`/`stylelintWarning` properties in a lint result object are not set when `styelint-disable` comment problems are reported.

Also, this change refactors the code around `styelint-disable` comments to make it more readable and prevent possible bugs.

The new code uses [`Result#warn()`](https://postcss.org/api/#result-warn) in PostCSS for a comment node, like reporting a normal rule problem.

> Which issue, if any, is this issue related to?

Closes #5049

> Is there anything in the PR that needs further explanation?

No, it's self-explanatory.

### CODE REVIEW

File: lib/reportDisables.mjs [hunk 1]
```diff
@@ -1,58 +1,43 @@
...
-		}
+			if (!reportDisablesForRule(configRules[rule])) continue;
 
-		for (const [rule, ranges] of Object.entries(rangeData)) {
-			for (const range of ranges) {
-				if (!reportDisablesForRule(config.rules[rule] || [])) continue;
-
-				// If the comment doesn't have a location, we can't report a useful error.
-				// In practice we expect all comments to have locations, though.
-				if (!range.comment.source || !range.comment.source.start) continue;
```
Comment (Mouvedia) [inline lib/reportDisables.mjs:38]: question

Do all comments have "locations"?

**Comment by Mouvedia** [comment]
> Also, this change refactors the code around styelint-disable comments to make it more readable and prevent possible bugs.

I think keeping the refactors in a separate commit would help the reviewers.

File: lib/reportDisables.mjs (hunk 1 above)
Comment (ybiquitous) [reply lib/reportDisables.mjs:38]: Do you mean the comments "// If the comment doesn't have a location, we can't report..."? If so, yes.

**Comment by ybiquitous** [comment]
> I think keeping these refactors in a separate commit would help the reviewers.

Sorry. I forgot to do that at this time. 🙏🏼

**Review by jeddy3 — APPROVED** [review]
LGTM, thank you!

**Pull request merged**